src/tarantool_schema.c

```diff
@@ -325,12 +349,75 @@ decode_index_parts_166(struct schema_field_value *parts, uint32_t part_count,
 		uint32_t len;
 		const char *str = mp_decode_str(data, &len);
 		part->field_type = field_type_by_name(str, len);
+		if (part->field_type == field_type_MAX)
+			return -1;
 
 		for (uint32_t j = 2; j < item_count; ++j)
 			mp_next(data);
-		/* Set default values. */
-		part->is_nullable = false;
-		part->coll_id = COLL_NONE;
+	}
+	return 0;
+}
+
+static int
+decode_index_part(struct schema_field_value *part, uint32_t map_size,
+		  const char **data)
+{
+	*part = field_val_def;
+	for (uint32_t i = 0; i < map_size; ++i) {
+		if (mp_typeof(**data) != MP_STR)
+			return -1;
+
+		uint32_t k_len;
+		const char *key = mp_decode_str(data, &k_len);
+		if (strncmp_strict(key, k_len, "type")) {
+			if (mp_typeof(**data) != MP_STR)
+				return -1;
+			uint32_t v_len;
+			const char *val = mp_decode_str(data, &v_len);
+			part->field_type = field_type_by_name(val, v_len);
+			if (part->field_type == field_type_MAX)
+				return -1;
+		} else if (strncmp_strict(key, k_len, "field")) {
+			if (mp_typeof(**data) != MP_UINT)
+				return -1;
+			part->field_number = mp_decode_uint(data);
+		} else if (strncmp_strict(key, k_len, "collation")) {
+			if (mp_typeof(**data) != MP_UINT)
+				return -1;
+			part->coll_id = mp_decode_uint(data);
+		} else if (strncmp_strict(key, k_len, "is_nullable")) {
+			if (mp_typeof(**data) != MP_BOOL)
+				return -1;
+			part->is_nullable = mp_decode_bool(data);
+		} else {
+			return -1;
+		}
+	}
+
+	if (part->coll_id != COLL_NONE &&
+	    part->field_type != FIELD_TYPE_STRING &&
+	    part->field_type != FIELD_TYPE_SCALAR) {
+		return -1;
+	}
```

Comment:
Is it some forgotten testing code? If not, I don't get it.

Reply:
It is from key_def.c. I will add a comment.

Reply:
> /* Collation is reasonable only for string and scalar parts. */

Okay, looks logical.